### PR TITLE
Remove IPackage and IPackageRepository implementations

### PR DESCRIPTION
**Pull request by joelverhagen**
Instead of depending on NuGet.Core interfaces for server-side concerns, remove the `IPackage` interface from `ServerPackage` and `IPackageRepository` from `ServerPackageRepository`. This allows the following:

1. Interface changes on `ServerPackage` without effecting NuGet.Core consumers `IPackage` interface.
1. Further deprecation of NuGet.Core. This means we can more easily move to 3.x/4.x libraries for NuGet.Server in the future.
1. Removal of unused properties and methods on `ServerPackage` and `ServerPackageRepository`.

NuGet.Core is still used to read .nupkgs from disk but is not part of the "core" server side API.

/cc @emgarten @skofman1 @dtivel

### CODE REVIEW

**Comment by joelverhagen** [comment]
Note that this PR is strictly a refactoring and should have no functional impact.

File: src/NuGet.Server.Core/Infrastructure/IdAndVersionEqualityComparer.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+using System;
+using System.Collections.Generic;
+using NuGet.Shared;
+
+namespace NuGet.Server.Core.Infrastructure
+{
+    public class IdAndVersionEqualityComparer : IEqualityComparer<IServerPackage>
+    {
+        private static IdAndVersionEqualityComparer _instance = new IdAndVersionEqualityComparer();
```
Comment (emgarten) [inline src/NuGet.Server.Core/Infrastructure/IdAndVersionEqualityComparer.cs]: nit: can be readonly

Comment (joelverhagen) [inline src/NuGet.Server.Core/Infrastructure/IdAndVersionEqualityComparer.cs]: Nice catch.

Comment (joelverhagen) [inline src/NuGet.Server.Core/Infrastructure/IdAndVersionEqualityComparer.cs]: Fixed.

**Pull request merged**